### ISSUE TITLE
Re-work docs site, possibly add support for versioned documentation

### ISSUE THREAD

**Issue by afolson** [meltano/meltano#5952]
Versioned docs (with version switcher - could use Mike for Mkdocs)

**Comment by tayloramurphy** [comment]
@afolson @aaronsteers what are some paths forward on this? This would be a big benefit both for the community and for our general release process as it would avoid situations where docs are updated prior to features being released and vice versa?

Is this something we could do in July or August?

**Comment by afolson** [comment]
@tayloramurphy This can be done in Jekyll with some work or we could move this to another tool like Sphinx. The latter might be better long term but is a larger undertaking and I'm not sure it's worth it just for this at the moment.

**Comment by tayloramurphy** [comment]
@afolson will you update this issue with your current status with Docasaurus? I'm realizing we've stated a lot of requirements for the new docs site but haven't written them down on the primary issue? 

Big things I believe we've aligned on are:

* versioned documentation
* 3 panel view (side bar, content, ToC)
* Easy search
* Good design and ease of maintenance

cc @DouweM @sbalnojan

**Comment by afolson** [comment]
Deployed: https://afolson-docs.netlify.app/concepts/

Ignore the weird side bar and default home page.

**Comment by afolson** [comment]
Other docs frameworks considered:

- Mkdocs
- Sphinx
- Staying with Jekyll
- Readme.io

All were ruled out due to the significant time it would take to re-theme them, time to create plugins to get functionality we wanted (specifically around docs versioning or OpenAPI/swagger tools), or lack of flexibility with diataxis framework.

**Comment by afolson** [comment]
@sbalnojan Yeah all of that will be added back. I haven't added any of the custom scripts/theme stuff yet.

**Comment by DouweM** [comment]
@afolson What are the steps yet to be done before we can replace the current docs site with this? I don't think it needs to _better_ yet, just have feature parity, and then we can continuously improve it from there.

**Comment by afolson** [comment]
Steps to get this to feature parity/launch:

- Improved search experience
- Home page

Ideally the theme would match the brand guidelines.

**Comment by afolson** [comment]
Given our limited bandwidth, I'm starting to feel like it makes sense to just make the current site work better for us. As nice as it would be to get versioned docs and prettier highlighting/TOC/etc out of the box, it's going to end up being a larger time investment than it's probably worth at this point in time.

@sbalnojan @tayloramurphy Would love to get your thoughts on this. Completely open to ideas. We can do versioning with Jekyll it's just not as pretty and may be a more manual process.

**Comment by tayloramurphy** [comment]
@afolson true we have limited bandwidth, but this is a high value improvement that is worth doing prior to GA. The docs are an extension of our product as as we move into Cloud GA we need to elevate the experience around the docs. 

What are the current blockers from your perspective? We'll have some frontend capacity with Wil. Is it mainly getting some engineering help on getting this deployed and replacing Jekyll? 

cc @DouweM

**Comment by afolson** [comment]
@tayloramurphy It sounds like most of Wil's time will be taken up with Cloud, Hub, and maybe some new website stuff as part of that launch. My worry is that it pushes this stuff out several more weeks. The new site needs a home page and some template tweaks to make it on par with what we currently have.

**Comment by tayloramurphy** [comment]
@afolson I'll sync up with @aaronsteers and @seth-meltano next week and see if we can get an estimate on scope given everything. Cloud is pending a few decisions on scope, Hub is awaiting the desigers for the bigger redesign. 

Do you have an open PR where the current status is?

**Comment by aaronsteers** [comment]
@afolson - can you provide the repo and a list of known remaining todo items for scoping?

**Comment by afolson** [comment]
@tayloramurphy @aaronsteers The site is [here](https://afolson-docs.netlify.app/) and deploys from [this repo](https://github.com/afolson/meltano-docs). I can make a PR for it if that's easier but initially didn't because it would have been a messier merge conflict later.

The homepage is the main thing to do to get it launched. I'm meeting with Wil tomorrow to discuss a wireframe that I made and will make specific issues with action items out of that.

**Comment by tayloramurphy** [comment]
@afolson would you mind inviting me to that call?

**Comment by afolson** [comment]
Docusaurus PR and action items here: https://github.com/meltano/meltano/pull/7208

Preview: https://deploy-preview-7208--meltano.netlify.app/

**Comment by DouweM** [comment]
Not a priority for now.

**Comment by tayloramurphy** [comment]
Closing as we're moving to docusaurus.
